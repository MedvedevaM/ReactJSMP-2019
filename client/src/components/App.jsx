import React, { Component } from 'react';
import { SearchPage } from './SearchPage.jsx';
import { ErrorBoundary } from './ErrorBoundary.jsx';

class App extends Component {
    render() {
        return (
            <ErrorBoundary>
                <SearchPage />
            </ErrorBoundary>
        );
    }
}

export default App;