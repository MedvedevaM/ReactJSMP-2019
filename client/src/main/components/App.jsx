import React from 'react';
import { hot } from 'react-hot-loader';
import SearchPage from './SearchPage.jsx';
import ErrorBoundary from './ErrorBoundary.jsx';

const App = ({ match: { params }, history }) => (
  <ErrorBoundary>
    <SearchPage history={history} chosenFilmId={params.id} searchValue={params.search} />
  </ErrorBoundary>
);

export default hot(module)(App);
