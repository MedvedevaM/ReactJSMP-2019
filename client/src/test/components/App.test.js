import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';
import App from "../../main/components/App.jsx";

describe("test app", () => {
    it("renders correctly", () => {
        const app = shallow(<App />);
        expect(toJson(app)).toMatchSnapshot();
    });
});
