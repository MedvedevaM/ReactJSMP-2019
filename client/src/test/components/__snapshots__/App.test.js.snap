// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app renders correctly 1`] = `
<ErrorBoundary>
  <SearchPage />
</ErrorBoundary>
`;
